documentation/modules/cruise-control/proc-generating-optimization-proposals.adoc

```diff
@@ -31,7 +37,58 @@ metadata:
     strimzi.io/cluster: my-cluster
 spec: {}
 ----
-
++
+You can also run the same type of rebalancing by specifying the `full` mode through the `spec.mode` field.
++
+.Example of full rebalancing by specifying the corresponding `full` mode
+[source,yaml,subs="attributes+"]
+----
+apiVersion: {KafkaRebalanceApiVersion}
+kind: KafkaRebalance
+metadata:
+  name: my-rebalance
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  mode: full
+----
++
+If you want to run the rebalance right after scaling up the Apache Kafka cluster and need to move already existing replicas to the newly added brokers, use the `add-brokers` mode.
++
+.Example of rebalancing to run after a scale up by using the `add-brokers` mode
+[source,yaml,subs="attributes+"]
+----
+apiVersion: {KafkaRebalanceApiVersion}
+kind: KafkaRebalance
+metadata:
+  name: my-rebalance
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  mode: add-brokers
+  brokers: [3, 4] <1>
+----
+<1> List of newly added brokers by the scale up operation. This is mandatory.
++
+If you want safely scale down the Apache Kafka cluster, consider to run a rebalance before to do so in order to move replicas off the brokers that are going to be removed.
+Use the `remove-brokers` mode in this case.
++
+.Example of rebalancing to run before a scale down by using the `remove-brokers` mode
+[source,yaml,subs="attributes+"]
+----
+apiVersion: {KafkaRebalanceApiVersion}
+kind: KafkaRebalance
+metadata:
+  name: my-rebalance
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  mode: remove-brokers
+  brokers: [3, 4] <1>
+----
+<1> List of brokers to be removed by the scale down operation. This is mandatory.
++
+All the steps following in this procedure are actually the same whatever is the mode you are using.
```

Comment:
"All the steps described in the following procedures are the same regardless of the rebalance mode you are using."